cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/task/AbstractClouderaManagerCommandListCheckerTask.java

```diff
@@ -0,0 +1,74 @@
+package com.sequenceiq.cloudbreak.cm.polling.task;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.cloudera.api.swagger.CommandsResourceApi;
+import com.cloudera.api.swagger.client.ApiException;
+import com.cloudera.api.swagger.model.ApiCommand;
+import com.cloudera.api.swagger.model.ApiHostRef;
+import com.sequenceiq.cloudbreak.cm.ClouderaManagerOperationFailedException;
+import com.sequenceiq.cloudbreak.cm.client.ClouderaManagerApiPojoFactory;
+import com.sequenceiq.cloudbreak.cm.polling.ClouderaManagerCommandListPollerObject;
+import com.sequenceiq.cloudbreak.structuredevent.event.CloudbreakEventService;
+
+public abstract class AbstractClouderaManagerCommandListCheckerTask<T extends ClouderaManagerCommandListPollerObject>
+        extends AbstractClouderaManagerApiCheckerTask<T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractClouderaManagerCommandListCheckerTask.class);
+
+    protected AbstractClouderaManagerCommandListCheckerTask(ClouderaManagerApiPojoFactory clouderaManagerApiPojoFactory,
+            CloudbreakEventService cloudbreakEventService) {
+        super(clouderaManagerApiPojoFactory, cloudbreakEventService);
+    }
+
+    protected boolean doStatusCheck(T pollerObject, CommandsResourceApi commandsResourceApi) throws ApiException {
+        List<ApiCommand> failedCommands = new ArrayList<>();
+        boolean commandsAreFinished = collectApiCommands(pollerObject, commandsResourceApi, failedCommands);
+        if (commandsAreFinished) {
+            processApiCommandResults(failedCommands);
+        }
+        return commandsAreFinished;
+    }
+
+    private boolean collectApiCommands(T pollerObject, CommandsResourceApi commandsResourceApi, List<ApiCommand> failedCommands)  throws ApiException {
```

Comment:
as we talked offline, please refactor this part

Reply:
no